src/Database/Type/DateTimeType.php

```diff
@@ -53,9 +53,12 @@ class DateTimeType extends Type implements TypeInterface
     /**
      * String format to use for DateTime parsing
      *
-     * @var string
+     * @var string|array
      */
-    protected $_format = 'Y-m-d H:i:s';
+    protected $_format = [
+        'Y-m-d H:i:s',
+        'Y-m-d\TH:i:s.uO',
```

Comment:
Are you saying that 5.7 fails to insert rows with​ the current datetime format marshalling?